CHANGELOG.md

```diff
@@ -1,3 +1,12 @@
+## 0.5.0 (31/12/2015)
+
+### Changes
+
+* The dependency to Guzzle is removed. We are now using PHP-HTTP to make us independent from any transport library. 
+* We are using PSR7 requests and responses
+* Introduced HttpFactory to create clients and responses
```

Comment:
It is not necessary, no. I did that for two reasons. First to put all discovery functions in one place. The rest of the lib does only contain references to the `HttpClient` and the `Plugin` interfaces.

The second reson why I prefer a HttpFactory is because it make the creating of a client simpler when you use plugins. See the [example in the readme](https://github.com/google/google-auth-library-php/pull/81/files#diff-04c6e90faac2675aa89e2176d2eec7d8R100) and also the [implementation](https://github.com/Nyholm/google-auth-library-php/blob/PSR7/src/Http/HttpFactory.php#L41) of `HttpFactory::getClient`.